server/src/test/java/org/eclipse/lsp/cobol/core/model/tree/statements/ObsoleteNodeTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.model.tree.statements;
+
+import com.google.common.collect.ImmutableList;
+import org.eclipse.lsp.cobol.core.messages.MessageTemplate;
+import org.eclipse.lsp.cobol.core.model.ErrorSeverity;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.tree.RemarksNode;
+import org.eclipse.lsp.cobol.core.model.tree.RootNode;
+import org.eclipse.lsp.cobol.core.semantics.NamedSubContext;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/** Test {@link ObsoleteNode} */
+class ObsoleteNodeTest {
+  @Test
+  public void testObsoleteNodeWarning() {
+    Locality locality = Locality.builder().build();
+    RootNode rootNode = new RootNode(locality, new NamedSubContext());
+    RemarksNode remarksNode = new RemarksNode(locality);
+    rootNode.addChild(remarksNode);
+
+    assertEquals(
+        rootNode.process(),
+        ImmutableList.of(
+            SyntaxError.syntaxError()
+                .severity(ErrorSeverity.WARNING)
+                .locality(locality)
+                .messageTemplate(
+                    MessageTemplate.of("This code block is deprecated and not supported"))
```

Comment:
You should not use the `MessageTemplate` here, but pass the string literal.

Reply:
.messageTemplate(MessageTemplate.of("cobolParser.ObsoleteCode")) ?